source/Octopus.Tentacle.Tests.Integration/CapabilitiesServiceV2Test.cs

```diff
@@ -56,21 +56,24 @@ public async Task CapabilitiesResponseShouldBeCached(TentacleConfigurationTestCa
             await using var clientAndTentacle = await tentacleConfigurationTestCase.CreateBuilder()
                 .WithPortForwarder(out var portForwarder)
                 .WithTentacleServiceDecorator(new TentacleServiceDecoratorBuilder()
-                    .HookServiceMethod<IAsyncClientCapabilitiesServiceV2, object, CapabilitiesResponseV2>(nameof(IAsyncClientCapabilitiesServiceV2.GetCapabilitiesAsync),
-                        null,
-                        async (_, response) =>
-                        {
-                            await Task.CompletedTask;
+                    .DecorateCapabilitiesServiceV2With(d => d
+                        .AfterGetCapabilities(
+                            async response =>
+                            {
+                                await Task.CompletedTask;
 
-                            capabilitiesResponses.Add(response);
+                                if (response != null)
```

Comment:
I added the null check here to handle the case where there is no response (due to an exception).

If it was `null` due to an exception, the exception will still be thrown by the service.